model/converter/json/from_domain.go

```diff
@@ -97,19 +97,26 @@ func (fd fromDomain) convertReferences(span *model.Span, preserveParentID bool)
 		length++
 	}
 	out := make([]json.Reference, 0, length)
-	if span.ParentSpanID != 0 && !preserveParentID {
-		out = append(out, json.Reference{
-			RefType: json.ChildOf,
-			TraceID: json.TraceID(span.TraceID.String()),
-			SpanID:  json.SpanID(span.ParentSpanID.String()),
-		})
-	}
+	var parentRefAdded bool
 	for _, ref := range span.References {
 		out = append(out, json.Reference{
 			RefType: fd.convertRefType(ref.RefType),
 			TraceID: json.TraceID(ref.TraceID.String()),
 			SpanID:  json.SpanID(ref.SpanID.String()),
 		})
+		if ref.TraceID == span.TraceID && ref.SpanID == span.ParentSpanID {
+			// Check if the parent reference already exists
+			parentRefAdded = true
+		}
+	}
+	if span.ParentSpanID != 0 && !preserveParentID && !parentRefAdded {
```

Comment:
not directly related to this diff, but I see that the only time `preserveParentID == true` happens is when this conversion is called from [ES storage WriteSpan function](https://github.com/jaegertracing/jaeger/blob/412baf60db217d6f41d7fdca04358069362b29ca/plugin/storage/es/spanstore/writer.go#L118). What was the reason why we needed to preserve ParentSpanID in ES? Same performance/storage optimization?